test/Interop/Cxx/class/type-classification-non-trivial-irgen.swift

```diff
@@ -0,0 +1,94 @@
+// RUN: %target-swift-frontend -enable-cxx-interop -I %S/Inputs %s -emit-ir | %FileCheck %s
+
+import TypeClassification
+
+// TODO: C++ objects with destructors should be tested here once we fully
+// support them.
+
+// CHECK-LABEL: define {{.*}}i1 @"$s4main37testStructWithCopyConstructorAndValueSbyF"
```

Comment:
Could you add an explanation of what these tests are verifying, so that future when IRGen patterns change people can update the CHECK lines appropriately, and not accidentally eliminate the part that we're thinking we are testing here.